runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -1704,10 +1719,11 @@ TR_J9SharedCacheServerVM::getClassFromSignature(const char * sig, int32_t sigLen
          }
       }
 
-   if (validated)
-      return j9class;
-   else
-      return NULL;
+   //if Validation fails still save NULL in the cache as it will remain invalidated in the future also
```

Comment:
The comment seems out of place here because we don't do any caching in the next few lines

Reply:
Note that the caching has already happened in `TR_J9ServerVM::getClassFromSignature(sig, sigLength, method, true);`
If you intended to store NULL when the validation didn't succeed, it's not happening.